selector/methods/distance.py

```diff
@@ -60,28 +61,43 @@ class MaxMin(SelectionBase):
     Structure‐Activity Relationships 21.6 (2002): 598-604.
     """
 
-    def __init__(self, fun_dist=None):
+    def __init__(self, fun_dist=None, ref_index="medoid"):
         """
         Initializing class.
 
         Parameters
         ----------
         fun_distance : callable
             Function for calculating the pairwise distance between sample points.
-            `fun_dist(X) -> X_dist` takes a 2D feature array of shape (n_samples, n_features)
+            `fun_dist(x) -> x_dist` takes a 2D feature array of shape (n_samples, n_features)
             and returns a 2D distance array of shape (n_samples, n_samples).
+        ref_index: int, str, list, optional
+            Index of the reference sample to start the selection algorithm from.
+            It can be an integer, a float, a list of integers or floats or a mixture of them,
+            or "medoid". When "medoid", the medoid center is chosen as the reference
+            sample. Default is "None".
```

Comment:
@FanwangM, To be consistent with other methods, we the `ref_index` should be an int and a list of integers with a default value of `None`, in which case mediod is used as the reference point. Being an index, I don't think we should support float. The user should pass the index in a proper type. Please fix other instances of `ref_index` based on this comment.